expression/distsql_builtin.go

```diff
@@ -41,103 +44,111 @@ func PbTypeToFieldType(tp *tipb.FieldType) *types.FieldType {
 	}
 }
 
-func getSignatureByPB(ctx sessionctx.Context, sigCode tipb.ScalarFuncSig, tp *tipb.FieldType, args []Expression) (f builtinFunc, e error) {
+func getSignatureByPB(ctx sessionctx.Context, sigCode tipb.ScalarFuncSig, tp *tipb.FieldType, metadata []byte, args []Expression) (f builtinFunc, e error) {
 	fieldTp := PbTypeToFieldType(tp)
 	base := newBaseBuiltinFunc(ctx, args)
 	base.tp = fieldTp
+	var message proto.Message
+	if len(metadata) != 0 {
+		err := proto.Unmarshal(metadata, message)
+		if err != nil {
+			logutil.BgLogger().Error("decode metadata", zap.Any("metadata", metadata), zap.Error(err))
+			return nil, nil
```

Comment:
Should we return the error here?

Reply:
In the [previous implementation](https://github.com/pingcap/tidb/blob/master/expression/expr_to_pb.go#L252-L259), we just return `nil`. @XuHuaiyu PTAL